### ISSUE TITLE
Get other property or index on select

### ISSUE THREAD

**Issue by Amerlander** [metonym/svelte-typeahead#6]
Is there any way to search in one property, but get another on select?

e.g. in this repl: https://svelte.dev/repl/3d17231bcce24ac681921968ea0d5d38?version=3.24.1
Is there a way to search in `state`, but to access the value of `abbr` on select?

Having the index  would also work, then one could do `states[index].abbr`. But since the index is only the index of the result and not the searched array position this doesnt work.

**Comment by Amerlander** [comment]
My actual workaround is searching in the array with the `selected` value: https://svelte.dev/repl/c709e73c2bec429c91a676c5cd52d89d?version=3.24.1

The downside of this is , that I cant search in multiple fields, since then the `selected` isn't a key anymore.

**Comment by metonym** [comment]
I think this should be possible.

In the dispatched event, we can include the `original` item that is selected as well as the `originalIndex`.

Example event payload from `on:select`:

```json
{
  "event": "select",
  "detail": {
    "selectedIndex": 2,
    "selected": "South Carolina",
    "original": {
      "id": 3,
      "state": "South Carolina"
    },
    "originalIndex": 3
  }
}
```

Would this work for your use case?

**Comment by Amerlander** [comment]
yes, including the original is exactly what I would need! :)

**Comment by metonym** [comment]
Published to NPM in v2.1.0.

[Updated Svelte REPL](https://svelte.dev/repl/a1b828d80de24f7e995b2365782c8d04?version=3.24.1)

Please let me know if this addresses your issue.

**Comment by Amerlander** [comment]
thank you, that works great for me. I just raised a PR for another little thing that I needed and I think one more will follow for filtering the results as a counterpart for extract.
